### PR TITLE
Private Key Download Button

### PR DESCRIPTION
**Pull request by DragonCoderz**
Made front end and back end changes to allow a prospective OSFCI user to download private keys to enable them to load their own firmware into the dev ci tool. 

Steps to Test Basic Functionality Works (Popup and Download Button appears, and making sure pop up doesn't show up again in your current login instance):

1) Open up the Dev CI homepage. Ensure that the download button or pop up does not appear.
2) Log in. After you complete this process, a pop up should appear along with a Download Private Keys button on the far right (to the left of the drop down). 
3) Click acknowledge to make sure the pop up goes away until next log in. You can ensure this by typing ``mylocalStorage['privKeyInfoAck']`` in the console. A value of 1 should be returned. You can test to make sure that the pop up doesn't show up by clicking the home button. 

Steps to make sure the download private key button actually sends the correct get request. (Complete these steps after doing the first section)

1) Click the download private key button. The private key binary file should either download or open up in a new tab. 

Step to Ensure Download Private Key Button goes away correctly

Situation 1: After logging out

1) Click the logout button. The private key should disappear until you re login.

Situation 2: When you run CI. 

1) move your mouse over your username, a drop down should show up. Click any interactive session. Once you get out of the waitlist and the session starts, the download button should disappear on the nav bar.

### CODE REVIEW

File: gateway/js/app.js [hunk 1]
```diff
@@ -888,6 +891,9 @@ function myAccount()
 
 function logged()
 {
+	if (typeof(mylocalStorage['privKeyInfoAck'] == 'undefined')) {
+		mylocalStorage['privKeyInfoAck'] = 0;
```
Comment (susmithp) [inline gateway/js/app.js:895]: We need to investigate on HTTP cookie to handle this. That we can cover in next PR.

File: gateway/server.go [hunk 2]
```diff
@@ -339,6 +340,18 @@ func home(w http.ResponseWriter, r *http.Request) {
 			base.Zlog.Fatalf("JSON encoding error: %s", err.Error())
 		}
 		w.Write([]byte(returnData))
+
+	case "get_private_key":
+		returnData, err := ioutil.ReadFile("/usr/local/production/keys/customer_private_key.pem")
```
Comment (susmithp) [inline gateway/server.go:345]: We can also think about moving this hardcoded file path into the config file.

**Review by susmithp — APPROVED** [review]
Requested some changes, but can be addressed in the next PR.

File: gateway/js/app.js (hunk 1 above)
Comment (arunkoshy) [reply gateway/js/app.js:895]: ack. session management overall will be overhauled in future.

**Review by arunkoshy — APPROVED** [review]
LGTM with caveats as discussed offline and on thread with @susmithp , @DragonCoderz

**Pull request merged**